package/lib/server.js

```diff
@@ -179,15 +182,15 @@ app.post(/^\/([^.]+)$/, (req, res) => {
 // Catch 404 and forward to error handler
 app.use((req, res, next) => {
   res.status(404)
-  res.render('404')
+  res.render('404.njk')
```

Comment:
So you don't need a separate engine defined for `njk` if templateExtension is set to `html`?

Reply:
I’ve learnt quite a bit about Express today!

* `app.set('view engine', ext)` sets the default file extension if no extension is given. If you set `ext` to `'foo'`, `res.render('thing')` will look for `thing.foo`, but `res.render('thing.bar')` will look for `thing.bar`. I’ve explicitly set the extension for 404 and 500 pages (which are included in the package) so the default value will be ignored.

* `app.engine(ext, callback)` sets the renderer to use for a given extension. However nunjucks.configure registers Nunjucks as the template engine for Express, so the extension value is ignored. We could replace this line with:

  ```diff
  - app.engine(templateExtension, getNunjucksEnv(app, env).render)
  + getNunjucksEnv(app, env)
  ```

  This might be a bit more explicit.

This set up means you can use both `.html` and `.njk` (or `.foo`) file extensions in a prototype if you’d want to, but you’d need to include file extensions in the `render()` method for any template that not using the default extension.

I think this is a good compromise. We can use `.html` for our example layouts (will open a separate PR when this is merged). For the prototype I’m using, I can set `templateExtension` config value to `njk` and the rig will continue working.